lib/util/package-rules/dep-patterns.ts

```diff
@@ -18,37 +16,26 @@ export class DepPatternsMatcher extends Matcher {
       return false;
     }
 
-    let isMatch = false;
-    for (const packagePattern of matchDepPatterns) {
-      const packageRegex = regEx(massagePattern(packagePattern));
-      if (packageRegex.test(depName)) {
-        logger.trace(`${depName} matches against ${String(packageRegex)}`);
-        isMatch = true;
-      }
-    }
-    return isMatch;
+    const massagedPatterns = matchDepPatterns.map((pattern) =>
+      pattern === '^*$' || pattern === '*' ? '**' : `/${pattern}/`,
+    );
```

Comment:
We should make this an option in `matchRegexOrGlobList`. 

added a PR for that https://github.com/renovatebot/renovate/pull/28552